lib/fact_changes.rb

```diff
@@ -118,55 +111,55 @@ def values_for_predicate(asset, predicate)
     (actual_values + values_to_add - values_to_destroy)
   end
 
-  def _build_fact_attributes(s, p, o, options = {})
-    t = [s,p,o, options]
-    params = { asset: t[0], predicate: t[1], literal: !(t[2].kind_of?(Asset)) }
-    params[:literal] ? params[:object] = t[2] : params[:object_asset] = t[2]
-    params = params.merge(t[3]) if t[3]
+  def _build_fact_attributes(asset, predicate, object, options = {})
+    literal = !(object.kind_of?(Asset))
+    params = { asset: asset, predicate: predicate, literal: literal }
+    literal ? params[:object] = object : params[:object_asset] = object
+    params = params.merge(options) if options
```

Comment:
This is mostly just naming thing.
I couldn't work out why we were generating an intermediate array (t) so got rid of it. I'm a bit worried I've missed something (https://fs.blog/chestertons-fence/) but it never seems to leave the scope of this function.

Reply:
The intermediate array was not needed.